src/OpenTelemetry.Api/Metrics/LabelSet.cs

```diff
@@ -29,11 +32,53 @@ public class LabelSet
         public LabelSet(IEnumerable<KeyValuePair<string, string>> labels)
         {
             this.Labels = labels;
+            this.LabelSetEncoded = this.GetLabelSetEncoded(labels);
         }
 
         /// <summary>
         /// Gets or sets the labels for this LabelSet.
         /// </summary>
         public IEnumerable<KeyValuePair<string, string>> Labels { get; set; }
+
+        /// <summary>
+        /// Gets or sets the label set in encoded form for this LabelSet.
+        /// </summary>
+        public string LabelSetEncoded { get; set; }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return this.LabelSetEncoded.GetHashCode();
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return this.LabelSetEncoded.Equals(((LabelSet)obj).LabelSetEncoded);
+        }
+
+        private string GetLabelSetEncoded(IEnumerable<KeyValuePair<string, string>> labels)
+        {
+            // Sort using keys.
+            var ordered = this.Labels.OrderBy(x => x.Key);
+            StringBuilder encoder = new StringBuilder();
+            bool isFirstLabel = true;
+
+            // simple encoding.
+            foreach (KeyValuePair<string, string> label in ordered)
+            {
+                if (!isFirstLabel)
+                {
+                    encoder.Append("\0");
```

Comment:
zero Unicode character? Why not something printe-able for better debugging?

Reply:
Opening new issue to track this separately. This implementation is a trivial one to temporarily make things work.